apps/manager-v2/src/app/manage-accounts/account-details/token-consumption/token-consumption.component.ts

```diff
@@ -0,0 +1,138 @@
+import { ChangeDetectionStrategy, Component, inject, OnInit } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import {
+  PaButtonModule,
+  PaDatePickerModule,
+  PaDropdownModule,
+  PaTableModule,
+  PaTabsModule,
+  PaTextFieldModule,
+} from '@guillotinaweb/pastanaga-angular';
+import { DropdownButtonComponent, InfoCardComponent, SisToastService } from '@nuclia/sistema';
+import { NucliaTokensMetric } from '@nuclia/core';
+import { isBefore, isFuture, subDays } from 'date-fns';
+import { AccountService } from '../../account.service';
+import { ManagerStore } from '../../../manager.store';
+import { BehaviorSubject, filter, map, switchMap, take } from 'rxjs';
+import { AccountDetails } from '../../account-ui.models';
+
+@Component({
+  selector: 'nma-token-consumption',
+  standalone: true,
+  imports: [
+    CommonModule,
+    PaTableModule,
+    DropdownButtonComponent,
+    PaDropdownModule,
+    PaDatePickerModule,
+    PaTextFieldModule,
+    InfoCardComponent,
+    PaButtonModule,
+    PaTabsModule,
+  ],
+  templateUrl: './token-consumption.component.html',
+  styleUrl: './token-consumption.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TokenConsumptionComponent implements OnInit {
+  private service = inject(AccountService);
+  private store = inject(ManagerStore);
+  private toaster = inject(SisToastService);
+
+  account = this.store.accountDetails;
+  kbList = this.store.kbList;
+
+  tokenConsumption = new BehaviorSubject<NucliaTokensMetric | null>(null);
+  selectedTab = 'account';
+
+  toDate: string;
+  toTime: number;
+  fromDate: string;
+  fromTime: number;
+  kbId?: string;
+
+  backup: { from: string; to: string };
+
+  constructor() {
+    const now = new Date();
+
+    // Range initialised with last 24h
+    this.toDate = now.toISOString();
+    this.toTime = now.getHours();
+
+    const from = subDays(now, 1);
+    this.fromDate = from.toISOString();
+    this.fromTime = from.getHours();
+
+    this.backup = {
+      from: `${this.from}`,
+      to: `${this.to}`,
+    };
+  }
+
+  ngOnInit() {
+    this.loadTokenConsumption();
+  }
+
+  loadTokenConsumption() {
+    this.account
+      .pipe(
+        filter((account) => !!account),
+        take(1),
+        map((accountDetails) => (accountDetails as AccountDetails).id),
+        switchMap((accountId) => this.service.loadTokenConsumption(accountId, this.from, this.to, this.kbId)),
```

Comment:
When accessing the usage page, I always get CORS error on the request to get the usage data. Is it expected ?

Reply:
yes, the endpoint is not on stage yet. This will work as soon as the backend is merged.